audio5.js

```diff
@@ -134,17 +134,29 @@
     trigger: function (evt) {
       if (this.channels && this.channels.hasOwnProperty(evt)) {
         var args = Array.prototype.slice.call(arguments, 1);
-        var a = [];
+
+        // temp array to store all subscribers for the channel evt that is being triggered that must be executed.
+        var evtSubscribers = [];
+
         while(this.channels[evt].length > 0) {
           var sub = this.channels[evt].shift();
-          if (typeof (sub.fn) === 'function') {
-            sub.fn.apply(sub.ctx, args);
-          }
+
           if ( !sub.once ){
-            a.push(sub);
+            // for any channel event triggers that are not marked as once only, we need to keep.
+            this.channels[evt].push(sub);
```

Comment:
Won't this line cause an infinite loop? If you shift the subscribe from the channel on line 142 and then push it again on line 146, the while loop in line 141 might never exit

Reply:
Yep - this was so shoddy refactoring on my behalf, I've opened a much more respectable pull request.  Apologies for your time used looking at this.

Reply:
No worries.. Let me know when I can review the code again

Reply:
https://github.com/zohararad/audio5js/pull/96

Here we go